File: Data_Structures/Maps_Hash_Dict/DivisiblePairCount.c
#include "stdio.h"
#include <stdlib.h>

int DivisiblePairsCount(int arr[], int n)
{
    int res = 0;
    for (int i = 0; i < n; ++i) {                             // Iterating through each pair.
       for (int j = i+1; j < n; ++j) {
            if (arr[i] % arr[j] == 0 || arr[j] % arr[i] == 0)              // Checking for problem condition.
              res++;
       }
    }
    return res;
}

int main()
{
    int n;
    int *arr;
    printf("Enter Size of the Array:  ");
    scanf("%d", &n);

    arr = malloc(sizeof(int) * n);

    printf("Enter the Array Elements:  ");
    for(int i = 0; i<n; i++)
    {
        scanf("%d",&arr[i]);
    }

    printf("Number of Divisible Pairs: %d \n", DivisiblePairsCount(arr, n));

    return 0;
}

File: Data_Structures/Maps_Hash_Dict/DivisiblePairCount.js
const readline = require('readline')
const reader = readline.createInterface(process.stdin, process.stdout); 

function countDivisiblePairs(arr) {
    let res = 0, n = arr.length
    for(let i=0;i<n;++i)
        for(let j=i+1;j<n;++j)
            if(arr[i]%arr[j] === 0 || arr[j]%arr[i] === 0)
                ++res
    return res
}

reader.question('Enter array elements seperated by space: ', arr => {
    let numbers = arr.split(" ")
    console.log("There are " + countDivisiblePairs(numbers) + " divisible pairs in given array.")
    reader.close()
})